src/environmentd/tests/sql.rs

```diff
@@ -3526,3 +3526,31 @@ fn test_peek_on_dropped_indexed_view() {
         })
         .unwrap();
 }
+
+/// Test AS OF in EXPLAIN. This output will only differ from the non-ASOF versions with RETAIN
+/// HISTORY, where the object and its indexes have differing compaction policies.
+#[mz_ore::test(tokio::test(flavor = "multi_thread", worker_threads = 1))]
+async fn test_explain_as_of() {
+    // TODO: This would be better in testdrive, but we'd need to support negative intervals in AS
+    // OF first.
+    let server = test_util::TestHarness::default().start().await;
+    let client = server.connect().await.unwrap();
+
+    // Retry until we are able to explain plan and timestamp for a few seconds ago.
+    Retry::default()
+        .clamp_backoff(Duration::from_secs(1))
+        .retry_async(|_| async {
+            let now: String = client
+                .query_one("SELECT mz_now()::text", &[])
+                .await
+                .unwrap()
+                .get(0);
+            let now: i64 = now.parse().unwrap();
+            let ts = now - 3000;
+            client.query_one(&format!("EXPLAIN TIMESTAMP FOR SELECT * FROM mz_internal.mz_cluster_replica_statuses AS OF {ts}"), &[]).await?;
+            client.query_one(&format!("EXPLAIN PLAN FOR SELECT * FROM mz_internal.mz_cluster_replica_statuses AS OF {ts}"), &[]).await?;
```

Comment:
Can you add a note that this works because `mz_cluster_replica_statuses` is a retained metrics table.